code/API_definitions/qos-booking-and-assignment.yaml

```diff
@@ -0,0 +1,1756 @@
+
+    DeviceIpv6Address:
+      description: |
+        The device should be identified by the observed IPv6 address, or by any single IPv6 address from within the subnet allocated to the device (e.g. adding ::0 to the /64 prefix).
+      type: string
+      format: ipv6
+      example: 2001:db8:85a3:8d3:1319:8a2e:370:7344
+
+    Status:
+      description: |
+        The current status of the requests in any of the endpoints in Booking or Assignment. The status can be one of the following:
+        * `PENDING` - The request is stil being processed. For booking, booking is requested and for assignements, the request is pending. 'statusInfo' may have additional information.
+        * `SUCCESSFUL` - The request is processed successfully. This may happen immediately.  The state will go to PENDING and then back to SUCCESSFUL in certain conditions.ƒ
+        * 'PARTIAL_SUCCESS' -  The request is processed with partial success - mostly applicable to device assignments.  'statusInfo' may have additional information. This enum is optional, and useful if the operator wants to communicate partical success.
+        * `FAILURE` - The request has been processed but failed. `statusInfo` may provide additional information about the reason for the unavailability.
+      enum:
+        - PENDING
+        - SUCCESSFUL
+        - PARTIAL_SUCCESS
+        - FAILURE
+
+    StatusChanged:
+      description: |
+        The current status of PENDING can be changed to SUCCESSFUL or PARTIAL_SUCCESS or FAILURE.  
+        In some situations (rare situations) a SUCCESSFUL booking or a device assignment may turn into a FAILURE because of natural disasters, catastrophic events, or for a unknown reason.
+        In such a situations 'statusInfo' may provide additional information about the reason for the unavailability
+      type: string
+      enum:
+        - SUCCESSFUL
+        - PARTIAL_SUCCESS
+        - FAILURE
+
+    BookingStatusInfo:
+      description: |
+        This schema provides a detailed status of a booking request
+        * `DURATION_EXPIRED` - Session terminated due to requested duration expired
+        * `NETWORK_TERMINATED` - Network terminated the QoS Booking for unknown reasons
+        * `DELETE_REQUESTED`- User requested the deletion of the QoS Booking
+        * 'VALIDATION_PENDING' - The validation is pending. The end user will get a callback with the availability of the booking
+        * 'SERVICE_NOT_AVAILABLE' - The requested booking could not be fullfilled because of operator could not support the service area/number of devices/qosProfile in the requested time
+        * 'QOS_PROFILE_NOT_SUPPORTED'  - The requested QoS Profile is not supported by the operator
+        * 'BOKING_UNKNOWN_ERROR' - An unknown error happend while booking
+        * 'BOOKING_COMPLETED' - The requested booking is commpleted successfully. 
+        * 'BOOKING_INVALID'  -  The given bookingId is invalid, malformed, or not available in the system

```

Comment:
As bookingId is a path param, this situation should end with a 404 error probably, instead of a 2xx with this value

Reply:
Any error in the backend when reported back to the client it should be 404 so this item is not going to be returned. It may be misleading.